pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -796,108 +793,61 @@ func createHostDnsmasqFile(ctx *zedrouterContext, bridge string) {
 	}
 }
 
-// Returns an IP address as a string, or "" if not found.
-func lookupOrAllocateIPv4(
-	ctx *zedrouterContext,
-	status *types.NetworkInstanceStatus,
-	mac net.HardwareAddr) (string, error) {
-
-	log.Functionf("lookupOrAllocateIPv4(%s-%s): mac:%s\n",
-		status.DisplayName, status.Key(), mac.String())
-	// Lookup to see if it exists
-	if ip, ok := status.IPAssignments[mac.String()]; ok {
+// Returns an IP address as a string, or "" on error
+func lookupOrAllocateIPv4(status *types.NetworkInstanceStatus,
+	appID uuid.UUID, appNum int, mac net.HardwareAddr) (string, error) {
+	log.Functionf("lookupOrAllocateIPv4(%s-%s): appNum:%d\n",
+		status.DisplayName, status.Key(), appNum)
+	if appNum == types.AppNumInvalid {
+		log.Functionf("(%s)no ip address\n", mac.String())
+		return "", nil
+	}
+	// if it exists
+	if ip := getIPAddrAssignment(status, mac); ip != nil {
 		log.Functionf("found Ip addr ( %s) for mac(%s)\n",
 			ip.String(), mac.String())
 		return ip.String(), nil
 	}
-
-	log.Functionf("bridgeName %s Subnet %v range %v-%v\n",
-		status.BridgeName, status.Subnet,
-		status.DhcpRange.Start.String(), status.DhcpRange.End.String())
-
 	if status.DhcpRange.Start == nil {
 		if status.Type == types.NetworkInstanceTypeSwitch {
-			log.Functionf("%s-%s switch means no bridgeIpAddr",
+			log.Functionf("%s-%s switch means no IpAddr",
```

Comment:
Presumably yetus will insist on this being capitalized IPAddr

Reply:
done.